### PR TITLE
fail validations on non-ignorable error codes

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/47631

## Description
fail validations on error codes which cannot be in ignored .pack-ignore. 

## Screenshots

example of how it looks like in the output (panorama xsoar supported):

image

while this is the panorama .pack-ignore 

image

## Must have
- [x] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

**Comment by GuyAfik** [comment]
@idovandijk this is not ready, ill let you know once it will be, therefore you have no reason to review.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## Unreleased
+* Added support in the **validate** command to fail validations on non-ignorable errors which are part of the .pack-ignore.
```
Comment (dorschw) [inline CHANGELOG.md:4]:
```suggestion
* Added a validation to the **validate** command, failing when non-ignorable errors are present in  .pack-ignore.
```

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 2]
```diff
@@ -26,49 +27,103 @@ class BaseValidator:
 
     def __init__(self, ignored_errors=None, print_as_warnings=False, suppress_print: bool = False,
                  json_file_path: Optional[str] = None):
+        # these are the ignored errors from the .pack-ignore including un-allowed error codes
         self.ignored_errors = ignored_errors if ignored_errors else {}
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:31]:
```suggestion
        self.ignored_errors = ignored_errors or {}
```

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 3]
```diff
@@ -26,49 +27,103 @@ class BaseValidator:
...
 
-        return False
+        return (
+            error_code in ignored_errors_pack_ignore or error_type in ignored_errors_pack_ignore
+        ) and (
+            error_code in ALLOWED_IGNORE_ERRORS
+        )
+
+    @staticmethod
+    def should_error_code_not_be_ignored_in_pack_ignore(error_code, ignored_errors_pack_ignore):
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:80]: what's the difference between this one and `should_ignore_error`?

File: demisto_sdk/commands/common/hook_validations/base_validator.py [hunk 4]
```diff
@@ -97,7 +158,7 @@ def formatted_error_str(error_type):
 
         formatted_error = formatted_error_str('ERROR')
         if should_print and not self.suppress_print:
-            if suggested_fix:
+            if suggested_fix and 'can not be ignored in .pack-ignore' not in formatted_error:
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/base_validator.py:161]: Not urgent, but using strings for conditionals is not the best practice, can we use some other flag instead?

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 5]
```diff
@@ -660,7 +661,7 @@ def test_get_error_ignore_list(self, mocker):
 
         validate_manager = ValidateManager()
         ignore_errors_list = validate_manager.get_error_ignore_list("fake")
-        assert ignore_errors_list['file_name'] == ['BA101', 'SC101', 'BA106']
+        assert ignore_errors_list['file_name'] == ['BA101', 'SC101', 'IN117', 'BA106', 'IN100']
```
Comment (dorschw) [inline demisto_sdk/commands/validate/tests/validators_test.py:664]: why is this modified?

**Review by dorschw — COMMENTED** [review]
looks great
scheduling us a mini demo before approval

File: demisto_sdk/commands/common/hook_validations/base_validator.py (hunk 3 above)
Comment (GuyAfik) [reply demisto_sdk/commands/common/hook_validations/base_validator.py:80]: basically the ```should_error_code_not_be_ignored_in_pack_ignore``` method checks out if an error code that is **in** the .pack-ignore should be ignored or not, while the ```should_ignore_error``` method defines if an error should be ignored in general (including errors from .pack-ignore and errors that can be ignored in parter/community supported/deprecated packs

File: demisto_sdk/commands/common/hook_validations/base_validator.py (hunk 4 above)
Comment (GuyAfik) [reply demisto_sdk/commands/common/hook_validations/base_validator.py:161]: you are right, ill handle it in a different way.

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 5 above)
Comment (GuyAfik) [reply demisto_sdk/commands/validate/tests/validators_test.py:664]: pay attention that in the ```validator_manager``` until now we collected **only** all the errors which can be ignored (we simply ignored from un-ignorable errors).

I changed the functionality in the way that we would get all the errors from the .pack-ignore including those who can not be ignored.

This is a test which checks that an un-ignorable errors won't be added to the ```ignore_errors``` list. 
hence modified it to include error codes which can not be ignored.
The logic is being handled in the ```base_validator``` now.

File: demisto_sdk/commands/common/hook_validations/base_validator.py (hunk 3 above)
Comment (dorschw) [reply demisto_sdk/commands/common/hook_validations/base_validator.py:80]: I see, thanks
perhaps a name like `is_error_allowed_in_pack_ignore` (keep in mind to revert the boolean result) will be clearer

Comment (GuyAfik) [reply demisto_sdk/commands/common/hook_validations/base_validator.py:82]: will be done!

**Review by dorschw — APPROVED** [review]
